core/modules/wsched/SchedulerBase.cc

```diff
@@ -65,4 +65,74 @@ void SchedulerBase::setPriorityDefault() {
 }
 
 
+int SchedulerBase::_incrCountForUserQuery(uint64_t queryId) {
+    std::pair<uint64_t, int> entry(queryId, 0);
+    std::lock_guard<std::mutex> lock(_countsMutex);
+    auto result = _userQueryCounts.insert(entry);
+    return ++result.first->second;
```

Comment:
I think I'm being paranoid about the first entry being 0. I looked around a little but couldn't verify that it would certainly  use 0 if there was no entry for queryId.